packages/api/src/controllers/logController.js

```diff
@@ -79,11 +79,50 @@ class logController extends baseController {
     }
   }
 
+  static getLogByActivityId(){
+    return async (req, res) => {
+      try{
+        if(req.params.activity_id){
+          const activity_id = req.params.activity_id
+          const log = await logServices.getLogById(activity_id);
+          res.json(log);
+        }else{
+          res.status(200).json([]);
+        }
+      }catch(exception){
+        const error = ExceptionHandler.handleException(exception);
+        res.status(error.status).json({ error:error.message });
+      }
+    }
+  }
+
+  static getLogByFarmId(){
+    return async (req, res) => {
+      try{
+        if(req.params.farm_id){
+          const farm_id = req.params.farm_id
+          const logs = await logServices.getLogByFarm(farm_id);
+          if(logs && !logs.length){
+            //TODO: replace [{}] with []
+            res.json([{}]);
```

Comment:
can we do this now? Or is there a front end restriction?

Reply:
fixed